### PR TITLE
ci: Workflow for code coverage

### PR DESCRIPTION
**Pull request by liya2017**
### What problem does this PR solve?

Problem Summary:

A workflow  for overage report, run ckb unit/integration test code coverage weekly and upload coverage report to codecov.io

### Release note 

```release-note
None: Exclude this PR from the release note.
```

### CODE REVIEW

**Review by keroro520 — CHANGES_REQUESTED** [review]
I suggest adding a `if` condition `github.event_name == 'schedule' && github.repository_owner == 'nervosnetwork'` to prevent redundant runnings on fork repos.

**Review by doitian — CHANGES_REQUESTED** [review]
- Delete `Related changes` and `Side effects` when they do not make sense.
- Select one from the three choices in `Release note`
- Do not paste links to private repository.

**Review by keroro520 — DISMISSED** [review]

**Comment by doitian** [comment]
- Please follow the convention of https://github.com/probot/semantic-pull-requests to name the PR title or at least one commit.
- Please provide "What problem does this PR solve?" and "What are changed?" briefly in this PR.

File: .github/workflows/coverage_report.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+          rustup toolchain list | { grep 'nightly' || rustup install nightly; }
+      - uses: actions/checkout@v2
+      - name: unit coverage
+        run: make cov
+      - name: integration cov
+        run: make integration-cov
+      - name: upload unit cov result to codecov.io
+        uses: codecov/codecov-action@v2
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```
Comment (doitian) [inline .github/workflows/coverage_report.yaml:26]: Not required for public repo [※](https://github.com/codecov/codecov-action#:~:text=%23%20not%20required%20for%20public%20repos)

File: .github/workflows/coverage_report.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+          files: ${{github.workspace}}/lcov-unit-test.info
+          flags: unit
+          override_commit: ${{github.sha}}
+      - name: upload integration cov result to codecov.io
+        uses: codecov/codecov-action@v2
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```
Comment (doitian) [inline .github/workflows/coverage_report.yaml:33]: Not required for public repo.

**Review by doitian — CHANGES_REQUESTED** [review]

File: .github/workflows/coverage_report.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+name: coverage_report
+
+on:
+  schedule:
+    - cron: '0 0 * * 5'
+
+jobs:
+  coverage_report:
+    name: coverage_report
+    runs-on: [self-hosted,Linux]
```
Comment (doitian) [inline .github/workflows/coverage_report.yaml:10]: According to the convention in https://talk.nervos.org/t/rfc-ckb-improve-ci-workflows/6053

```suggestion
    runs-on: self-hosted-coverage-report-ubuntu-??
```

For `??`, I'm not sure which version we are using on the self runners.

Comment (liya2017) [reply .github/workflows/coverage_report.yaml]: our self runner version is 20.04

File: .github/workflows/coverage_report.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+name: coverage_report
+
+on:
+  schedule:
+    - cron: '0 0 * * 5'
+
+jobs:
+  coverage_report:
+    name: coverage_report
+    runs-on: [self-hosted,self-hosted-coverage_report-ubuntu-20.04]
```
Comment (doitian) [inline .github/workflows/coverage_report.yaml:10]:
```suggestion
    runs-on: self-hosted-coverage_report-ubuntu-20.04
```

**Review by keroro520 — APPROVED** [review]

**Review by doitian — APPROVED** [review]

File: .github/workflows/coverage_report.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
+name: coverage_report
+
+on:
+  schedule:
```
Comment (doitian) [inline .github/workflows/coverage_report.yaml:4]: Better to add a `workflow_dispatch` so we can generate the report on demand.

File: .github/workflows/coverage_report.yaml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    - cron: '0 0 * * 5'
+
+jobs:
+  coverage_report:
+    name: coverage_report
+    runs-on: self-hosted-coverage_report-ubuntu-20.04
+    if: |
+        github.event_name == 'schedule' &&
```
Comment (doitian) [inline .github/workflows/coverage_report.yaml:12]: It's enough to check only `github.repository_owner == 'nervosnetwork'`.

```suggestion
```

**Review by doitian — CHANGES_REQUESTED** [review]

File: .github/workflows/coverage_report.yaml (hunk 5 above)
Comment (liya2017) [reply .github/workflows/coverage_report.yaml:4]: added

**Review by doitian — APPROVED** [review]

**Comment by doitian** [comment]
bors r=doitian,keroro520

**Pull request merged**